org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTParser.java

```diff
@@ -1229,68 +1233,28 @@ private ASTNode internalCreateASTCached(IProgressMonitor monitor) {
 						throw new IllegalStateException();
 					}
 					if ((this.bits & CompilationUnitResolver.PARTIAL) != 0) {
-						searcher = new NodeSearcher(this.focalPointPosition);
```

Comment:
Small note on this code: rather than pass around the NodeSearcher, I have opted to pass the focalPointPosition (or -1) around, as NodeSearcher is not API and should not be in any interfaces.